ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -121,16 +139,57 @@ object ParallelIndexerFactory {
                   },
               )
 
-      def subscribe(readService: ReadService): Future[Source[Unit, NotUsed]] =
-        dbDispatcher
-          .executeSql(metrics.daml.parallelIndexer.initialization)(storageBackend.initialize)
-          .map(initialized =>
-            ingest(initialized.lastEventSeqId.getOrElse(0L))(
-              readService.stateUpdates(beginAfter = initialized.lastOffset)
+      def subscribe(resourceContext: ResourceContext)(readService: ReadService): Handle = {
+        implicit val rc: ResourceContext = resourceContext
+        implicit val ec: ExecutionContext = resourceContext.executionContext
+        implicit val matImplicit: Materializer = mat
+        haCoordinator.protectedExecution { connectionInitializer =>
+          val killSwitchCaptor = Promise[KillSwitch]()
```

Comment:
```suggestion
          val killSwitchPromise = Promise[KillSwitch]()
```
As there s an actual class under that name I would use the word promise to aid readability.

Reply:
good point. changed